compass-js-frontend/src/main/webapp/ui/view/editor/hierarchy.js

```diff
@@ -52,10 +52,32 @@ var HierarchyView = AmpersandView.extend({
 		});
 		// what we actually want to save is the internal tree object.
 		this.tree = $tree.fancytree('getTree');
+		$tree.on('contextmenu', this.showContextMenu.bind(this));
+	},
+	registerContextOnMenu: function() {
+		var $menu = $('.basicContextContainer');
+		$menu.on('contextmenu', function() {
+			basicContext.close();
+			return false;
+		});
+	},
+	showContextMenu: function(event, data){
+		var node = $.ui.fancytree.getNode(event);
+		this.selectCurrentNode(node);
+		var items = [
+			{ type: 'item', title: 'Add Node', icon: 'ion-plus-round', fn: this.newChild.bind(this) },
+			{ type: 'item', title: 'Delete Node', icon: 'ion-minus-round', fn: function() {} }
+		]
+		basicContext.show(items,event);
```

Comment:
basicContext should be scoped to this module, similar to jQuery, i.e. `var basicContext = global.basicContext`, to make it clear that this is a browser-level bower import.